### ISSUE TITLE
Output error from an MLP model 

### ISSUE THREAD

**Issue by geekco2000** [deeplearning4j/deeplearning4j#3230]
When I am trying to print the output of an MLP regression model I got some errors (train set : 70 000, test set : 30 000):
[ ?, ?, ?, ?, ...] --> it doesn't print correctly (weird characters), looks like something is missing, but the rmse looks very good, what this means ?
Otherwise, when I train my model (for example) in a train set of 15 observations and a test set of 5 observations, then I can see the output printed correctly, but the rmse isn't good at all (which is quite normal).
val predicted = net.output(test.getFeatures)
println("*** PREDICTED ***")
println(predicted)
[12 367, 15 768, 11 676, ...] --> looks like this

Also when I change the updater parameter from : 
      .updater(Updater.NESTEROVS)
      .momentum(0.9) 
To 
	  .rmsDecay(0.95)
	  .updater(Updater.RMSPROP)
Then I can print correctly the output error.

val net = sparkNet.fit(trainData)
val labels = test.getLabels
println("*** LABELS ***")
println(labels)
val predicted = net.output(test.getFeatures)
println("*** PREDICTED ***")
println(predicted)

Is there a bug or I missed something ?

Thank you !

**Comment by raver119** [comment]
"weird characters" means NaN actually.

**Comment by geekco2000** [comment]
Ok, sorry but I don't really get it ? Does it means something is missing in my neural net configuration ? 
Or maybe I added something I shouldn't ? 

Thank you !!

**Comment by tomthetrainer** [comment]
nan's occur when a value is so close to 0 that it can not be represented

see https://en.wikipedia.org/wiki/NaN

You can think of it in this case a placeholder for a very small value. 

I am closing this, if you have further questions please re-open.

**Comment by raver119** [comment]
NaNs in your case basically means - your model was broken during training. I.e. some underflow happened. Or something like that.
